include/klee/Module/KModule.h

```diff
@@ -131,33 +126,33 @@ struct KReturnBlock : KBlock {
 struct KFunction : public KCallable {
 private:
   std::unordered_map<std::string, KBlock *> labelMap;
+  const unsigned globalIndex;
 
 public:
   KModule *parent;
   llvm::Function *function;
-
-  unsigned numArgs, numRegisters;
-  unsigned id;
-
-  std::unordered_map<unsigned, KInstruction *> registerToInstructionMap;
-  unsigned numInstructions;
-  unsigned numBlocks;
   KInstruction **instructions;
 
-  bool kleeHandled = false;
+  [[nodiscard]] KInstruction *getInstructionByRegister(size_t reg) const;
 
   std::unordered_map<const llvm::Instruction *, KInstruction *> instructionMap;
+  // TODO maybe change to std::unique_ptr<KBlock[]>
```

Comment:
Then you will need to clear the memory allocated for KBlocks.

Reply:
Deleted this comment.
It was comment about ways to optimize, but this one doesn't get real profit.